Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/AzureStorageUri.java

```diff
@@ -115,6 +133,21 @@ public String relativize(AzureStorageUri target) {
         return relativePath;
     }
 
+    /**
+     * Encode the relative path in the Azure Storage URI
+     * @param rawPath un-encoded raw path in the Azure Storage URI
+     * @return encoded path in the Azure Storage URI
+     */
+    public static String encodePath(String rawPath) {
```

Comment:
I suppose we don't need this method any more. Refer to the code ` .resolve("./" + target)`

Reply:
Let's try this:

```java
String encodedPath = URI.create("/").relativize(new URIBuilder().setPath("/" + rawPath).build())
return rawPath.startsWith("/") ? "/" + encodedPath : encodedPath
```